core/shared/src/main/scala/scalaz/zio/Void.scala

```diff
@@ -0,0 +1,10 @@
+package scalaz.zio
+
+/**
+  * Represents something that can't exist
+  */
+abstract final class Void {
```

Comment:
The name `Void` is a bit overused IMO, scalaz has it and even there is one in java which already caused some confusion. While using zio as an standalone lib will be no issue, but when using it together with scalaz base it will be confusing.

I suggest, since this inhabited type will be used only to signal the absence of errors, that we call it `NoError` (or anything similar)

WDYT @jdegoes ?

Reply:
I agree `Void` is horribly overused. The real problem is that it's called `Void` in other languages, such as Haskell and PureScript.

My idea is that Scalaz will no longer have its own void. Instead, it will inherit the void from this package.

Reply:
In that case, shouldn't we port `VoidModule` from Scalaz? In more general cases (with variance involved) `Void` is better being bottom type and this naive implementation is not replacement for Scalaz's one. If scalaz/scalaz#1874 gets merged, having both `Void`s will be problematic.

Reply:
I agree regarding porting `Void` from Scalaz. @jdegoes what's your opinion?

Reply:
Although now I see that it has some dependencies from the rest of `Scalaz`, like `As` and `Is` (https://github.com/scalaz/scalaz/blob/series/8.0.x/base/shared/src/main/scala/scalaz/core/void.scala#L60). I have to admin I don't fully understand `Scalaz` `Void` implementation and I'm not sure what to do with those dependencies.

Reply:
I do no think we should import it. It does not actually infer as a bottom type. We can create `Is` and `As` instances for this `Void` in Scalaz 8.